### PR TITLE
Add cloud connector button to inventory

### PR DESCRIPTION
**Pull request by MariaAga**
Without actual functionality until https://github.com/theforeman/foreman_rh_cloud/pull/451/files is ready
Also changed the sync button style to be pf4

Screenshot from 2021-01-27 11-14-42
Screenshot from 2021-01-27 11-07-40
Screenshot from 2021-01-27 11-06-24

### CODE REVIEW

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { CloudConnectorButton } from './CloudConnectorButton';
+import { configureCloudConnector } from './CloudConnectorActions';
+import { selectStatus, selectJobLink } from './CloudConnectorSelectors';
+
+const ConnectedCloudConnectorButton = () => {
+  const status = useSelector(store => selectStatus(store));
+  const jobLink = useSelector(store => selectJobLink(store));
```
Comment (MariaAga) [inline webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/index.js:9]: Can change to `useSelector(selectStatus);`

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorActions.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { post } from 'foremanReact/redux/API';
+import { CONFIGURE_CLOUD_CONNECTOR } from './CloudConnectorConstants';
+import { inventoryUrl } from '../../../../ForemanInventoryHelpers';
+
+export const configureCloudConnector = () =>
+  post({
+    key: CONFIGURE_CLOUD_CONNECTOR,
+    url: inventoryUrl('enable_cloud_connector'),
+  });
```
Comment (laviro) [inline webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorActions.js:9]: it didn't actually fire the API middleware action, maybe a missing `dispatch`?

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorSelectors.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import {
+  CONFIGURE_CLOUD_CONNECTOR,
+  CONNECTOR_STATUS,
+} from './CloudConnectorConstants';
+
+export const selectStatus = state =>
+  selectAPIResponse(state, CONFIGURE_CLOUD_CONNECTOR).task_status ||
+  CONNECTOR_STATUS.NOT_RESOLVED;
+
```
Comment (laviro) [inline webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorSelectors.js:11]: I think we also need to track the status of the API post request itself

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import { selectStatus, selectJobLink } from './CloudConnectorSelectors';
+
+const ConnectedCloudConnectorButton = () => {
+  const status = useSelector(selectStatus);
+  const jobLink = useSelector(selectJobLink);
+  const dispatch = useDispatch();
+  return (
+    <CloudConnectorButton
+      status={status}
+      onClick={() => dispatch(configureCloudConnector)}
```
Comment (laviro) [inline webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/index.js:14]: for now, it doesn't fire the `configureCloudConnector` on click

File: webpack/__mocks__/foremanReact/redux/API/index.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
-export default {
+export const post = data => ({ type: 'post-some-type', ...data });
+export const API = {
```
Comment (laviro) [inline webpack/__mocks__/foremanReact/redux/API/index.js:2]: maybe we should update all operations to the structure you added?

```suggestion
export default {
 post: data => ({ type: 'post-some-type', ...data });
}
```

though should we really test its functionality? I think it's more of a responsibility of Foreman

**Review by laviro — COMMENTED** [review]
Thanks @MariaAga ! left a few comments

File: webpack/__mocks__/foremanReact/redux/API/index.js (hunk 5 above)
Comment (MariaAga) [reply webpack/__mocks__/foremanReact/redux/API/index.js:2]: I like to use it to see that the action sent the correct params, without this the action test will show `undefined`. I can add the rest of the operations (except delete) to this structure

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/index.js (hunk 4 above)
Comment (MariaAga) [reply webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/index.js]: Changed to `configureCloudConnector()`

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorConstants.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export const CONFIGURE_CLOUD_CONNECTOR = 'CONFIGURE_CLOUD_CONNECTOR';
+export const CONNECTOR_STATUS = {
+  PENDING: 'PENDING',
+  RESOLVED: 'RESOLVED',
+  NOT_RESOLVED: 'RESOLVED',
```
[the remark this answers is not in the document]
Comment (MariaAga) [reply webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorConstants.js]: Yup

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorActions.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { post } from 'foremanReact/redux/API';
+import { CONFIGURE_CLOUD_CONNECTOR } from './CloudConnectorConstants';
+import { inventoryUrl } from '../../../../ForemanInventoryHelpers';
+
+export const configureCloudConnector = () =>
+  post({
+    key: CONFIGURE_CLOUD_CONNECTOR,
+    url: inventoryUrl('enable_cloud_connector'),
```
Comment (laviro) [inline webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorActions.js:8]:
```suggestion
    url: inventoryUrl('cloud_connector'),
```

to match Erics work

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorActions.js [hunk 8]
```diff
@@ -1,9 +1,27 @@
...
+    url: inventoryUrl('cloud_connector'),
+    successToast: response => (
+      <span>
+        {__('Cloud connector setup is in progress now: ')}
+        <a href={foremanUrl(`/job_invocations/${response.data.id}`)}>
+          {__('Cloud connector job link')}
+        </a>
+      </span>
+    ),
+    errorToast: error => `cloud connector setup failed: ${error}`,
```
Comment (laviro) [inline webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorActions.js:20]: can you add a translation?

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorActions.js [hunk 9]
```diff
@@ -1,9 +1,27 @@
...
 
 export const configureCloudConnector = () =>
   post({
     key: CONFIGURE_CLOUD_CONNECTOR,
-    url: inventoryUrl('enable_cloud_connector'),
+    url: inventoryUrl('cloud_connector'),
+    successToast: response => (
+      <span>
+        {__('Cloud connector setup is in progress now: ')}
+        <a href={foremanUrl(`/job_invocations/${response.data.id}`)}>
```
[the remark this answers is not in the document]
Comment (MariaAga) [reply webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorActions.js:15]: If we can I'm not sure how

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorActions.js (hunk 8 above)
Comment (MariaAga) [reply webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorActions.js:20]: Added

File: webpack/ForemanInventoryUpload/Components/InventoryFilter/InventoryFilterReducer.js [hunk 10]
```diff
@@ -37,7 +37,7 @@ export default (
     }
     case LAYOUT_INITIALIZE: {
       // Layout action changed in Jul 20 2020 - https://github.com/theforeman/foreman/commit/e4c39a7d8f8b50ba45ef63e46f6f6914b69f247a
-      const { title } = organization; // org was renamed
+      const { title } = organization || {}; // org was renamed
```
Comment (MariaAga) [inline webpack/ForemanInventoryUpload/Components/InventoryFilter/InventoryFilterReducer.js:40]: The plugin pages didn't load without adding this as I was getting `organization=undefined`

**Pull request merged**